aodntools/timeseries_products/hourly_timeseries.py

```diff
@@ -0,0 +1,539 @@
+
+            ## keep TIME as the only index
+            df_temp = df_temp.reset_index().set_index('TIME')
+            df_temp = df_temp[parameter_names]
+
+            df_temp = PDresample_by_hour(df_temp, function_dict, function_stats)  # do the magic
+            df_temp['instrument_index'] = np.repeat(file_index, len(df_temp)).astype('int32')
+            df_data = pd.concat([df_data, df_temp.reset_index()], ignore_index=True, sort=False)
+
+        file_index += 1
+
+    df_metadata.index.rename('INSTRUMENT', inplace=True)
+    df_data.index.rename('OBSERVATION', inplace=True)
+    ## rename index to TIME
+    df_data.rename(columns={'index': 'TIME'}, inplace=True)
+
+
+    nc_metadata = xr.Dataset({'LONGITUDE': (['INSTRUMENT'], df_metadata['LONGITUDE'].astype('float32')),
+                              'LATITUDE': (['INSTRUMENT'], df_metadata['LATITUDE'].astype('float32')),
+                              'NOMINAL_DEPTH': (['INSTRUMENT'], df_metadata['NOMINAL_DEPTH'].astype('float32')),
+                              'instrument_id': (['INSTRUMENT'], df_metadata['instrument_id'].astype('|S256')),
+                              'source_file': (['INSTRUMENT'], df_metadata['source_file'].astype('|S256'))})
+
+    nc_data = xr.Dataset.from_dataframe(df_data)
+    nc_aggregated = xr.merge([nc_metadata, nc_data])
+
+
+    ## add global attributes
+    add_attribute = {'rejected_files': "\n".join(list(bad_files))}
+    nc_aggregated.attrs = set_globalattr(nc_aggregated, globalattr_file, site_code, add_attribute, parameter_names)
+
+
+    ## add variable attributes
+    variablenames_others = ['TIME', 'LONGITUDE', 'LATITUDE', 'NOMINAL_DEPTH',
+                            'instrument_index', 'instrument_id', 'source_file']
+    add_variable_attribute = {'PRES_REL': {'applied_offset_by_instrument': applied_offset}}
+    parameter_names_all = list(set(parameter_names_all))
+    variable_attributes = variable_attribute_dictionary
+
+    time_units = variable_attributes['TIME'].pop('units')
+    time_calendar = variable_attributes['TIME'].pop('calendar')
+
+    ## add attributes to TIME, LAT/LON, and index variables
+    for variable in variablenames_others:
+        nc_aggregated[variable].attrs = variable_attributes[variable]
+
+    for variable in parameter_names_all:
+        ancillary_variables_attr = []
+        nc_aggregated[variable].attrs = variable_attributes[variable]
+        nc_aggregated[variable].attrs['long_name'] = function_dict[variable] + " " + nc_aggregated[variable].attrs['long_name']
+        nc_aggregated[variable].attrs.update({'cell_methods': 'TIME:' + function_dict[variable] + ' (interval: 1 hr comment: time mid point)'})
+        for stat_method in function_stats:
+            variable_stat_name = variable + "_" + stat_method
+            ancillary_variables_attr += [variable_stat_name]
+
+            if stat_method == 'count':
+                nc_aggregated[variable_stat_name].attrs['units'] = 1
```

Comment:
`units` attribute must be a string, i.e. `"1"`

Reply:
fixed